### ISSUE TITLE
When setting a custom `results-dir`, flank fails to find CostReport.txt

### ISSUE THREAD

**Issue by runningcode** [Flank/flank#751]
This is happening in flank 8.1.0.
I set the `results-dir` to `fladle-master-1089-runningcode-PR15052-2020-04-29-16.46.08.723`.
```
[2020-04-29T15:11:53.724Z] java.nio.file.NoSuchFileException: results/fladle-master-1089-runningcode-PR15052-2020-04-29-16.46.08.723/CostReport.txt
[2020-04-29T15:11:53.724Z] 	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
[2020-04-29T15:11:53.724Z] 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
[2020-04-29T15:11:53.724Z] 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
[2020-04-29T15:11:53.724Z] 	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
[2020-04-29T15:11:53.724Z] 	at java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:434)
[2020-04-29T15:11:53.724Z] 	at java.nio.file.Files.newOutputStream(Files.java:216)
[2020-04-29T15:11:53.724Z] 	at java.nio.file.Files.write(Files.java:3292)
[2020-04-29T15:11:53.724Z] 	at ftl.util.Utils.write(Utils.kt:26)
[2020-04-29T15:11:53.724Z] 	at ftl.reports.CostReport.write(CostReport.kt:42)
[2020-04-29T15:11:53.724Z] 	at ftl.reports.CostReport.run(CostReport.kt:48)
[2020-04-29T15:11:53.724Z] 	at ftl.reports.util.ReportManager.generate(ReportManager.kt:104)
[2020-04-29T15:11:53.724Z] 	at ftl.run.TestRunner.newRun(TestRunner.kt:349)
[2020-04-29T15:11:53.724Z] 	at ftl.run.TestRunner$newRun$1.invokeSuspend(TestRunner.kt)
[2020-04-29T15:11:53.724Z] 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.ResumeModeKt.resumeUninterceptedMode(ResumeMode.kt:45)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.internal.ScopeCoroutine.afterCompletionInternal(Scopes.kt:32)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.JobSupport.completeStateFinalization(JobSupport.kt:310)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.JobSupport.tryFinalizeFinishingState(JobSupport.kt:236)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.JobSupport.tryMakeCompletingSlowPath(JobSupport.kt:849)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:811)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:787)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:111)
[2020-04-29T15:11:53.724Z] 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:270)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:79)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:54)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:36)
[2020-04-29T15:11:53.724Z] 	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
[2020-04-29T15:11:53.724Z] 	at ftl.cli.firebase.test.android.AndroidRunCommand.run(AndroidRunCommand.kt:49)
[2020-04-29T15:11:53.724Z] 	at picocli.CommandLine.executeUserObject(CommandLine.java:1729)
[2020-04-29T15:11:53.724Z] 	at picocli.CommandLine.access$900(CommandLine.java:145)
[2020-04-29T15:11:53.724Z] 	at picocli.CommandLine$RunLast.handle(CommandLine.java:2101)
[2020-04-29T15:11:53.724Z] 	at picocli.CommandLine$RunLast.handle(CommandLine.java:2068)
[2020-04-29T15:11:53.724Z] 	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:1935)
[2020-04-29T15:11:53.724Z] 	at picocli.CommandLine.execute(CommandLine.java:1864)
[2020-04-29T15:11:53.724Z] 	at ftl.Main$Companion.main(Main.kt:41)
[2020-04-29T15:11:53.724Z] 	at ftl.Main.main(Main.kt)
```

**Comment by pawelpasterz** [comment]
Please, check with snapshot version

**Comment by pawelpasterz** [comment]
@runningcode I am unable to reproduce this error on latest snapshot version. I think that might be fixed when flank moved to API based results.
cc @bootstraponline

**Comment by runningcode** [comment]
Hey, it turns out that there was a `/` in the `results-dir` argument. Can we make flank fail fast when the `results-dir` is invalid?

**Comment by pawelpasterz** [comment]
I think this is reasonable to have feature like this. To clarify, `/` was in the middle of `results-dir` argument?

**Comment by runningcode** [comment]
yeah

**Comment by pawelpasterz** [comment]
This bug will be fixed with #753 . Feel free to add your comments there.

**Comment by runningcode** [comment]
Perfect. thanks.
